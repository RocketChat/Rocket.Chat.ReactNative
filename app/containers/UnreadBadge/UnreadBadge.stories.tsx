import React from 'react';
import { View } from 'react-native';

import { themes } from '../../lib/constants';
import UnreadBadge from '.';
import { ThemeContext, TSupportedThemes } from '../../theme';

export default {
	title: 'Unread Badge'
};

const StoryTester = ({ children }: { children: React.ReactElement | React.ReactElement[] }) => (
	<View
		style={{
			flex: 1,
			flexDirection: 'row',
			alignItems: 'center',
			justifyContent: 'space-evenly'
		}}
	>
		{children}
	</View>
);

export const All = () => (
	<StoryTester>
		<UnreadBadge unread={9} small />
		<UnreadBadge unread={999} small />
		<UnreadBadge unread={9} />
		<UnreadBadge unread={9999} />
		<UnreadBadge unread={9} userMentions={1} />
		<UnreadBadge unread={9} groupMentions={1} />
		<UnreadBadge unread={9} tunread={[1]} />
	</StoryTester>
);

export const Small = () => (
	<StoryTester>
		<UnreadBadge unread={9} small />
		<UnreadBadge unread={999} small />
	</StoryTester>
);

export const Normal = () => (
	<StoryTester>
		<UnreadBadge unread={9} />
		<UnreadBadge unread={9999} />
	</StoryTester>
);

export const DifferentMentionTypes = () => (
	<StoryTester>
		<UnreadBadge unread={1} />
		<UnreadBadge unread={1} userMentions={1} />
		<UnreadBadge unread={1} groupMentions={1} />
		<UnreadBadge unread={1} userMentions={1} groupMentions={1} />
		<UnreadBadge unread={1} tunread={[1]} />
		<UnreadBadge unread={1} tunreadUser={[1]} />
		<UnreadBadge unread={1} tunreadGroup={[1]} />
	</StoryTester>
);

const ThemeStory = ({ theme }: { theme: TSupportedThemes }) => (
	<ThemeContext.Provider value={{ theme, colors: themes[theme] }}>
		<StoryTester>
			<UnreadBadge unread={1} />
			<UnreadBadge unread={1} userMentions={1} />
			<UnreadBadge unread={1} groupMentions={1} />
			<UnreadBadge tunread={[1]} />
		</StoryTester>
	</ThemeContext.Provider>
);

export const Themes = () => (
	<>
		<ThemeStory theme='light' />
		<ThemeStory theme='dark' />
		<ThemeStory theme='black' />
	</>
);
