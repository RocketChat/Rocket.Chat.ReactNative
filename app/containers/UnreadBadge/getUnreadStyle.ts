import { IUnreadBadge } from '.';
import { themes } from '../../lib/constants/colors';
import { TSupportedThemes } from '../../theme';

interface IGetUnreadStyle extends Omit<IUnreadBadge, 'small' | 'style'> {
	theme: TSupportedThemes;
}

export const getUnreadStyle = ({
	unread,
	userMentions,
	groupMentions,
	theme,
	tunread,
	tunreadUser,
	tunreadGroup
}: IGetUnreadStyle) => {
	if ((!unread || unread <= 0) && !tunread?.length) {
		return {};
	}

	let backgroundColor = themes[theme].unreadColor;
	const color = themes[theme].buttonText;
	if ((userMentions && userMentions > 0) || tunreadUser?.length) {
		backgroundColor = themes[theme].mentionMeColor;
	} else if ((groupMentions && groupMentions > 0) || tunreadGroup?.length) {
		backgroundColor = themes[theme].mentionGroupColor;
	} else if (tunread && tunread?.length > 0) {
		backgroundColor = themes[theme].tunreadColor;
	}

	return {
		backgroundColor,
		color
	};
};
