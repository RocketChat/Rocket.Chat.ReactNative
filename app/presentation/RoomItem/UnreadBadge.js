import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';

import styles from './styles';
import { themes } from '../../constants/colors';

const UnreadBadge = React.memo(({
	theme, unread, userMentions, type
}) => {
	if (!unread || unread <= 0) {
		return;
	}
	if (unread >= 1000) {
		unread = '999+';
	}
	const mentioned = userMentions > 0 && type !== 'd';

	return (
		<View
			style={[
				styles.unreadNumberContainer,
				{ backgroundColor: mentioned ? themes[theme].tintColor : themes[theme].borderColor }
			]}
		>
			<Text
				style={[
					styles.unreadText,
					{ color: mentioned ? themes[theme].buttonText : themes[theme].bodyText }
				]}
			>{ unread }
			</Text>
		</View>
	);
});

UnreadBadge.propTypes = {
	theme: PropTypes.string,
	unread: PropTypes.number,
	userMentions: PropTypes.number,
	type: PropTypes.string
};

export default UnreadBadge;
