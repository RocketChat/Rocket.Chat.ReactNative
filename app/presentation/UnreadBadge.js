import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, StyleSheet } from 'react-native';

import sharedStyles from '../views/Styles';
import { themes } from '../constants/colors';

const styles = StyleSheet.create({
	unreadNumberContainer: {
		minWidth: 21,
		height: 21,
		paddingVertical: 3,
		paddingHorizontal: 5,
		borderRadius: 10.5,
		alignItems: 'center',
		justifyContent: 'center',
		marginLeft: 10
	},
	unreadText: {
		overflow: 'hidden',
		fontSize: 13,
		...sharedStyles.textMedium,
		letterSpacing: 0.56,
		textAlign: 'center'
	}
});

const UnreadBadge = React.memo(({
	theme, unread, userMentions, groupMentions, style
}) => {
	if (!unread || unread <= 0) {
		return;
	}
	if (unread >= 1000) {
		unread = '999+';
	}

	let backgroundColor = themes[theme].unreadBackground;
	const color = themes[theme].buttonText;
	if (userMentions > 0) {
		backgroundColor = themes[theme].mentionMeColor;
	} else if (groupMentions > 0) {
		backgroundColor = themes[theme].mentionGroupColor;
	}

	return (
		<View
			style={[
				styles.unreadNumberContainer,
				{ backgroundColor },
				style
			]}
		>
			<Text
				style={[
					styles.unreadText,
					{ color }
				]}
			>{unread}
			</Text>
		</View>
	);
});

UnreadBadge.propTypes = {
	theme: PropTypes.string,
	unread: PropTypes.number,
	userMentions: PropTypes.number,
	groupMentions: PropTypes.number,
	style: PropTypes.object
};

export default UnreadBadge;
