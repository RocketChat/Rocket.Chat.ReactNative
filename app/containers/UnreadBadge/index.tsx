import React from 'react';
import { StyleProp, StyleSheet, Text, View, ViewStyle } from 'react-native';

import sharedStyles from '../../views/Styles';
import { getUnreadStyle } from './getUnreadStyle';
import { useTheme } from '../../theme';

const styles = StyleSheet.create({
	unreadNumberContainerNormal: {
		height: 21,
		paddingVertical: 3,
		paddingHorizontal: 5,
		borderRadius: 10.5,
		alignItems: 'center',
		justifyContent: 'center',
		marginLeft: 10
	},
	unreadNumberContainerSmall: {
		borderRadius: 10.5,
		alignItems: 'center',
		justifyContent: 'center'
	},
	unreadText: {
		fontSize: 13,
		...sharedStyles.textSemibold
	},
	textSmall: {
		fontSize: 10
	}
});

export interface IUnreadBadge {
	unread?: number;
	userMentions?: number;
	groupMentions?: number;
	style?: StyleProp<ViewStyle>;
	tunread?: any[];
	tunreadUser?: any[];
	tunreadGroup?: any[];
	small?: boolean;
	hideUnreadStatus?: boolean;
	hideMentionStatus?: boolean;
}

const UnreadBadge = React.memo(
	({
		unread,
		userMentions,
		groupMentions,
		style,
		tunread,
		tunreadUser,
		tunreadGroup,
		small,
		hideMentionStatus,
		hideUnreadStatus
	}: IUnreadBadge) => {
		const { theme } = useTheme();

		if ((!unread || unread <= 0) && !tunread?.length) {
			return null;
		}

		if (hideUnreadStatus && hideMentionStatus) {
			return null;
		}

		// Return null when hideUnreadStatus is true and isn't a direct mention
		if (hideUnreadStatus && !((userMentions && userMentions > 0) || tunreadUser?.length)) {
			return null;
		}

		const { backgroundColor, color } = getUnreadStyle({
			theme,
			unread,
			userMentions,
			groupMentions,
			tunread,
			tunreadUser,
			tunreadGroup
		});

		if (!backgroundColor) {
			return null;
		}
		let text: any = unread || tunread?.length;
		if (small && text >= 100) {
			text = '+99';
		}
		if (!small && text >= 1000) {
			text = '+999';
		}
		text = text.toString();

		let minWidth = 21;
		if (small) {
			minWidth = 11 + text.length * 5;
		}

		return (
			<View
				style={[
					small ? styles.unreadNumberContainerSmall : styles.unreadNumberContainerNormal,
					{ backgroundColor, minWidth },
					style
				]}
			>
				<Text style={[styles.unreadText, small && styles.textSmall, { color }]} numberOfLines={1}>
					{text}
				</Text>
			</View>
		);
	}
);

export default UnreadBadge;
