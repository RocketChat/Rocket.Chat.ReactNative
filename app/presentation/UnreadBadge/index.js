import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, StyleSheet } from 'react-native';

import sharedStyles from '../../views/Styles';
import { getUnreadStyle } from './getUnreadStyle';
import { withTheme } from '../../theme';

const styles = StyleSheet.create({
	unreadNumberContainerNormal: {
		height: 21,
		paddingVertical: 3,
		paddingHorizontal: 5,
		borderRadius: 10.5,
		alignItems: 'center',
		justifyContent: 'center',
		marginLeft: 10
	},
	unreadNumberContainerSmall: {
		borderRadius: 10.5,
		alignItems: 'center',
		justifyContent: 'center'
	},
	unreadText: {
		fontSize: 13,
		...sharedStyles.textSemibold
	},
	textSmall: {
		fontSize: 10
	}
});

const UnreadBadge = React.memo(({
	theme, unread, userMentions, groupMentions, style, tunread, tunreadUser, tunreadGroup, small
}) => {
	if ((!unread || unread <= 0) && (!tunread?.length)) {
		return;
	}
	const { backgroundColor, color } = getUnreadStyle({
		theme, unread, userMentions, groupMentions, tunread, tunreadUser, tunreadGroup
	});

	if (!backgroundColor) {
		return null;
	}
	let text = unread || tunread?.length;
	if (small && text >= 100) {
		text = '+99';
	}
	if (!small && text >= 1000) {
		text = '+999';
	}
	text = text.toString();

	let minWidth = 21;
	if (small) {
		minWidth = 11 + text.length * 5;
	}

	return (
		<View
			style={[
				small ? styles.unreadNumberContainerSmall : styles.unreadNumberContainerNormal,
				{ backgroundColor, minWidth },
				style
			]}
		>
			<Text
				style={[
					styles.unreadText,
					small && styles.textSmall,
					{ color }
				]}
				numberOfLines={1}
			>{text}
			</Text>
		</View>
	);
});

UnreadBadge.propTypes = {
	theme: PropTypes.string,
	unread: PropTypes.number,
	userMentions: PropTypes.number,
	groupMentions: PropTypes.number,
	style: PropTypes.object,
	tunread: PropTypes.array,
	tunreadUser: PropTypes.array,
	tunreadGroup: PropTypes.array,
	small: PropTypes.bool
};

export default withTheme(UnreadBadge);
