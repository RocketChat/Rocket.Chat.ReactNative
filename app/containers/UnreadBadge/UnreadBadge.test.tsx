import { generateSnapshots } from '../../../.rnstorybook/generateSnapshots';
import * as stories from './UnreadBadge.stories';

generateSnapshots(stories);
