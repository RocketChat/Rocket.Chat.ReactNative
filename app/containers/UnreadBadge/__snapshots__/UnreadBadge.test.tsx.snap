// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Snapshots: All should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-evenly",
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
        },
        {
          "backgroundColor": "#9EA2A8",
          "borderRadius": 10.5,
          "minWidth": 16,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          {
            "fontSize": 10,
          },
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      9
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
        },
        {
          "backgroundColor": "#9EA2A8",
          "borderRadius": 10.5,
          "minWidth": 26,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          {
            "fontSize": 10,
          },
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      +99
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 10,
          "paddingHorizontal": 5,
          "paddingVertical": 3,
        },
        {
          "backgroundColor": "#9EA2A8",
          "borderRadius": 10.5,
          "minWidth": 21,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          undefined,
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      9
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 10,
          "paddingHorizontal": 5,
          "paddingVertical": 3,
        },
        {
          "backgroundColor": "#9EA2A8",
          "borderRadius": 10.5,
          "minWidth": 21,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          undefined,
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      +999
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 10,
          "paddingHorizontal": 5,
          "paddingVertical": 3,
        },
        {
          "backgroundColor": "#F5455C",
          "borderRadius": 10.5,
          "minWidth": 21,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          undefined,
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      9
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 10,
          "paddingHorizontal": 5,
          "paddingVertical": 3,
        },
        {
          "backgroundColor": "#F38C39",
          "borderRadius": 10.5,
          "minWidth": 21,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          undefined,
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      9
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 10,
          "paddingHorizontal": 5,
          "paddingVertical": 3,
        },
        {
          "backgroundColor": "#095AD2",
          "borderRadius": 10.5,
          "minWidth": 21,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          undefined,
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      9
    </Text>
  </View>
</View>
`;

exports[`Story Snapshots: DifferentMentionTypes should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-evenly",
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 10,
          "paddingHorizontal": 5,
          "paddingVertical": 3,
        },
        {
          "backgroundColor": "#9EA2A8",
          "borderRadius": 10.5,
          "minWidth": 21,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          undefined,
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      1
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 10,
          "paddingHorizontal": 5,
          "paddingVertical": 3,
        },
        {
          "backgroundColor": "#F5455C",
          "borderRadius": 10.5,
          "minWidth": 21,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          undefined,
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      1
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 10,
          "paddingHorizontal": 5,
          "paddingVertical": 3,
        },
        {
          "backgroundColor": "#F38C39",
          "borderRadius": 10.5,
          "minWidth": 21,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          undefined,
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      1
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 10,
          "paddingHorizontal": 5,
          "paddingVertical": 3,
        },
        {
          "backgroundColor": "#F5455C",
          "borderRadius": 10.5,
          "minWidth": 21,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          undefined,
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      1
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 10,
          "paddingHorizontal": 5,
          "paddingVertical": 3,
        },
        {
          "backgroundColor": "#095AD2",
          "borderRadius": 10.5,
          "minWidth": 21,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          undefined,
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      1
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 10,
          "paddingHorizontal": 5,
          "paddingVertical": 3,
        },
        {
          "backgroundColor": "#F5455C",
          "borderRadius": 10.5,
          "minWidth": 21,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          undefined,
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      1
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 10,
          "paddingHorizontal": 5,
          "paddingVertical": 3,
        },
        {
          "backgroundColor": "#F38C39",
          "borderRadius": 10.5,
          "minWidth": 21,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          undefined,
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      1
    </Text>
  </View>
</View>
`;

exports[`Story Snapshots: Normal should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-evenly",
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 10,
          "paddingHorizontal": 5,
          "paddingVertical": 3,
        },
        {
          "backgroundColor": "#9EA2A8",
          "borderRadius": 10.5,
          "minWidth": 21,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          undefined,
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      9
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 10,
          "paddingHorizontal": 5,
          "paddingVertical": 3,
        },
        {
          "backgroundColor": "#9EA2A8",
          "borderRadius": 10.5,
          "minWidth": 21,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          undefined,
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      +999
    </Text>
  </View>
</View>
`;

exports[`Story Snapshots: Small should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-evenly",
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
        },
        {
          "backgroundColor": "#9EA2A8",
          "borderRadius": 10.5,
          "minWidth": 16,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          {
            "fontSize": 10,
          },
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      9
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
        },
        {
          "backgroundColor": "#9EA2A8",
          "borderRadius": 10.5,
          "minWidth": 26,
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
            "textAlign": "left",
          },
          {
            "fontSize": 10,
          },
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      +99
    </Text>
  </View>
</View>
`;

exports[`Story Snapshots: Themes should match snapshot 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginLeft": 10,
            "paddingHorizontal": 5,
            "paddingVertical": 3,
          },
          {
            "backgroundColor": "#9EA2A8",
            "borderRadius": 10.5,
            "minWidth": 21,
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "backgroundColor": "transparent",
              "fontFamily": "Inter",
              "fontSize": 13,
              "fontWeight": "600",
              "textAlign": "left",
            },
            undefined,
            {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        1
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginLeft": 10,
            "paddingHorizontal": 5,
            "paddingVertical": 3,
          },
          {
            "backgroundColor": "#F5455C",
            "borderRadius": 10.5,
            "minWidth": 21,
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "backgroundColor": "transparent",
              "fontFamily": "Inter",
              "fontSize": 13,
              "fontWeight": "600",
              "textAlign": "left",
            },
            undefined,
            {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        1
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginLeft": 10,
            "paddingHorizontal": 5,
            "paddingVertical": 3,
          },
          {
            "backgroundColor": "#F38C39",
            "borderRadius": 10.5,
            "minWidth": 21,
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "backgroundColor": "transparent",
              "fontFamily": "Inter",
              "fontSize": 13,
              "fontWeight": "600",
              "textAlign": "left",
            },
            undefined,
            {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        1
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginLeft": 10,
            "paddingHorizontal": 5,
            "paddingVertical": 3,
          },
          {
            "backgroundColor": "#095AD2",
            "borderRadius": 10.5,
            "minWidth": 21,
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "backgroundColor": "transparent",
              "fontFamily": "Inter",
              "fontSize": 13,
              "fontWeight": "600",
              "textAlign": "left",
            },
            undefined,
            {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        1
      </Text>
    </View>
  </View>,
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginLeft": 10,
            "paddingHorizontal": 5,
            "paddingVertical": 3,
          },
          {
            "backgroundColor": "#9EA2A8",
            "borderRadius": 10.5,
            "minWidth": 21,
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "backgroundColor": "transparent",
              "fontFamily": "Inter",
              "fontSize": 13,
              "fontWeight": "600",
              "textAlign": "left",
            },
            undefined,
            {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        1
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginLeft": 10,
            "paddingHorizontal": 5,
            "paddingVertical": 3,
          },
          {
            "backgroundColor": "#B43C4C",
            "borderRadius": 10.5,
            "minWidth": 21,
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "backgroundColor": "transparent",
              "fontFamily": "Inter",
              "fontSize": 13,
              "fontWeight": "600",
              "textAlign": "left",
            },
            undefined,
            {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        1
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginLeft": 10,
            "paddingHorizontal": 5,
            "paddingVertical": 3,
          },
          {
            "backgroundColor": "#955828",
            "borderRadius": 10.5,
            "minWidth": 21,
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "backgroundColor": "transparent",
              "fontFamily": "Inter",
              "fontSize": 13,
              "fontWeight": "600",
              "textAlign": "left",
            },
            undefined,
            {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        1
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginLeft": 10,
            "paddingHorizontal": 5,
            "paddingVertical": 3,
          },
          {
            "backgroundColor": "#245399",
            "borderRadius": 10.5,
            "minWidth": 21,
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "backgroundColor": "transparent",
              "fontFamily": "Inter",
              "fontSize": 13,
              "fontWeight": "600",
              "textAlign": "left",
            },
            undefined,
            {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        1
      </Text>
    </View>
  </View>,
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginLeft": 10,
            "paddingHorizontal": 5,
            "paddingVertical": 3,
          },
          {
            "backgroundColor": "#9EA2A8",
            "borderRadius": 10.5,
            "minWidth": 21,
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "backgroundColor": "transparent",
              "fontFamily": "Inter",
              "fontSize": 13,
              "fontWeight": "600",
              "textAlign": "left",
            },
            undefined,
            {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        1
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginLeft": 10,
            "paddingHorizontal": 5,
            "paddingVertical": 3,
          },
          {
            "backgroundColor": "#B43C4C",
            "borderRadius": 10.5,
            "minWidth": 21,
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "backgroundColor": "transparent",
              "fontFamily": "Inter",
              "fontSize": 13,
              "fontWeight": "600",
              "textAlign": "left",
            },
            undefined,
            {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        1
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginLeft": 10,
            "paddingHorizontal": 5,
            "paddingVertical": 3,
          },
          {
            "backgroundColor": "#955828",
            "borderRadius": 10.5,
            "minWidth": 21,
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "backgroundColor": "transparent",
              "fontFamily": "Inter",
              "fontSize": 13,
              "fontWeight": "600",
              "textAlign": "left",
            },
            undefined,
            {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        1
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginLeft": 10,
            "paddingHorizontal": 5,
            "paddingVertical": 3,
          },
          {
            "backgroundColor": "#245399",
            "borderRadius": 10.5,
            "minWidth": 21,
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "backgroundColor": "transparent",
              "fontFamily": "Inter",
              "fontSize": 13,
              "fontWeight": "600",
              "textAlign": "left",
            },
            undefined,
            {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        1
      </Text>
    </View>
  </View>,
]
`;
